pkg/cloud/azure/actuators/cluster/actuator.go

```diff
@@ -132,7 +136,12 @@ func (a *Actuator) Delete(cluster *clusterv1.Cluster) error {
 		}
 	*/
 
-	if err := resourcesSvc.DeleteResourceGroup(); err != nil {
+	groupsSvc, err := groups.NewService(scope)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create groups service %q", cluster.Name)
+	}
+
+	if err := groupsSvc.Delete(context.TODO()); err != nil {
```

Comment:
Looks like we need to passing `ctx` appropriately. Shouldn't we be having a `ctx` variable in the actuator scope ?

At the very very least, I would suggest the following :

```suggestion
	if err := groupsSvc.Delete(context.Background()); err != nil {
```

Reply:
Yep! i want to address it in another PR, where we define a top level context and then that controls how long the reconcile loop can run, i dont want to mix the context PR into this, just bloats the whole scope :(